Source/Test/SystematicTesting/TestingEngine.cs

```diff
@@ -576,8 +576,10 @@ private bool RunNextIteration(uint iteration)
                 }
 
                 // Cleans up the runtime before the next iteration starts.
+                runtimeLogger?.Clear();
                 runtimeLogger?.Dispose();
                 runtime?.Dispose();
+                GC.Collect();
```

Comment:
I am not sure why we need to call the GC directly here, since these objects will go out of scope in the next iteration and a GC will get triggered even without us doing it. I would prefer to avoid calling this low level API directly, unless we have clear data that shows this indeed helps?

Reply:
yep, fair enough, I was mostly just playing with GC.collect here to see if it would make the memory performance profile more flat and less major saw tooth going on.  But we can remove it if you feel strongly about it.  I look at it as a "hint" to the garbage collector that this is a great time to do a collect since we just finished an iteration.

Reply:
Yea, i was hopping we only do the min necessary changes in this PR to get this use case PWS/leak down, in a quick local experiment it only seems that the StringBuilder change is necessary together with that `StackTrace` disposal I did locally that I have not pushed to your branch (perhaps I should so we keep in one branch, its very small).

This change seems to not have any impact (or if it does it could be nondeterministic and up to the GC scheduler). Another reason is that our RF is currently experimenting with leak-detection techniques and might end up requiring to rely on explicit GCs in the end of the test iteration to check if finalizers/disposals have been called, so wanted to avoid us calling this method until he gets time to try deeply these APIs (as he might end up removing it anyway).

Reply:
sure go ahead and party on my branch so we can both work on this together.  I'll remove this GC call.

Reply:
Thanks man! Just pushed the change that disposes the `ScheduleTrace`.